packages/steiger/src/features/calculate-diagnostic-severities/calculate-final-severity.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { describe, expect, it, vi } from 'vitest'
+
+import { joinFromRoot, parseIntoFsdRoot } from '../../_lib/prepare-test'
+import calculateFinalSeverities from './calculate-final-severity'
+import { GlobGroupWithSeverity } from '../../models/config'
+
+vi.mock('../../models/config', async () => {
+  const ruleToGlobs: Record<string, Array<GlobGroupWithSeverity>> = {
+    rule1: [
+      { files: [], ignores: [], severity: 'warn' },
```

Comment:
And related to that discussion, I think we should remove empty arrays here
```suggestion
      { severity: 'warn' },
```